src/boot/chid.c

```diff
@@ -57,10 +57,19 @@ typedef struct SmbiosInfo {
         char16_t *smbios_fields[_CHID_SMBIOS_FIELDS_MAX];
 } SmbiosInfo;
 
-static void smbios_info_populate(SmbiosInfo *ret_info) {
+/* Only exposed for testing purposes */
+void smbios_info_populate_internal(SmbiosInfo *ret_info, bool reset);
```

Comment:
Why not to declare this in chid.h ??

Reply:
Because it shouldn't be used for anything except tests.

There are the following approaches:
1. Declare it only where used (which is what I did)
2. Declare it as `void smbios_info_populate_internal(void *ret_info, bool reset)` with a comment like "do NOT use anywhere except for testing"
3. Drop caching altogether (will make things a bit slower if `chid_match` is called multiple times, i.e. for add-ons or different device types once those will be added).

I'm not sure what's the best approach here. What do you think should chosen?

Reply:
Typically, we choose 2 at several places. So, if there is no special reasons to 'hide' the function, please declare it in the header with a short comment.